sdk/core/azure-core/src/main/java/com/azure/core/util/polling/ChainedPollingStrategy.java

```diff
@@ -0,0 +1,101 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A polling strategy that chains multiple polling strategies, finds the first strategy that can poll the current
+ * long running operation, and polls with that strategy.
+ *
+ * @param <T> the {@link TypeReference} of the response type from a polling call, or BinaryData if raw response body
+ *            should be kept
+ * @param <U> the {@link TypeReference} of the final result object to deserialize into, or BinaryData if raw response
+ *            body should be kept
+ */
+public class ChainedPollingStrategy<T, U> implements PollingStrategy<T, U> {
+    private final List<PollingStrategy<T, U>> pollingStrategies;
+    private PollingStrategy<T, U> pollableStrategy = null;
+
+    /**
+     * Creates an empty chained polling strategy.
+     */
+    public ChainedPollingStrategy() {
+        this.pollingStrategies = new ArrayList<>();
+    }
+
+    /**
+     * Creates a chained polling strategy with 3 known polling strategies, {@link OperationResourcePollingStrategy},
+     * {@link LocationPollingStrategy}, and {@link StatusCheckPollingStrategy}, in this order. The first strategy that
+     * can poll on the initial response will be used. The created chained polling strategy is capable of handling most
+     * of the polling scenarios in Azure.
+     *
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     * @param <T> the {@link TypeReference} of the response type from a polling call, or BinaryData if raw response body
+     *            should be kept
+     * @param <U> the {@link TypeReference} of the final result object to deserialize into, or BinaryData if raw response
+     *            body should be kept
+     * @return the initialized chained polling strategy with the default chain
+     */
+    public static <T, U> ChainedPollingStrategy<T, U> createDefault(
+            HttpPipeline httpPipeline,
+            Context context) {
+        return new ChainedPollingStrategy<T, U>()
+            .addPollingStrategy(new OperationResourcePollingStrategy<>(httpPipeline, context))
+            .addPollingStrategy(new LocationPollingStrategy<>(httpPipeline, context))
+            .addPollingStrategy(new StatusCheckPollingStrategy<>());
+    }
+
+    /**
+     * Adds a polling strategy to the chain of polling strategies.
+     * @param pollingStrategy the polling strategy to add
+     * @return the modified ChainedPollingStrategy instance
+     */
+    public ChainedPollingStrategy<T, U> addPollingStrategy(PollingStrategy<T, U> pollingStrategy) {
+        this.pollingStrategies.add(pollingStrategy);
+        return this;
+    }
+
+    @Override
+    public Mono<Boolean> canPoll(Response<?> initialResponse) {
+        return Flux.fromIterable(pollingStrategies)
+            .concatMap(strategy -> strategy.canPoll(initialResponse).doOnNext(canPoll -> {
```

Comment:
Can more than one strategy return `true` for `canPoll()`? If yes, there might be a race condition here since the doOnNext method is a side-effect method and doesn't prevent the concatMap from evaluating the next strategy.

Reply:
The `any()` on line 77 will subscribe to only the first pollable strategy.

Reply:
Fixed.